### PR TITLE
pubsub: make connection log less aggressively

### LINKED ISSUE #2614: pubsub: streaming connection logs too aggressively
First reported [here](https://github.com/GoogleCloudPlatform/google-cloud-java/issues/1827#issuecomment-343760699).

The connection logs a warning even though the subscriber is shutting down. When shutting down, this error is normal. The subscriber waits to process all messages, but by the time this happens, another message might have already arrived. Trying to ack this message could error: it races with us trying to shutdown streaming channel.

This is sub-optimal but "OK". The message will be redelivered to another machine. The fix for this is to not log such errors when the the subscriber is shutting down.

### PR DESCRIPTION
**Pull request by pongad**
Fixes #2614

### CODE REVIEW

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java [hunk 1]
```diff
@@ -275,7 +275,9 @@ public void sendAckOperations(
         requestObserver.onNext(request);
       }
     } catch (Exception e) {
-      logger.log(Level.WARNING, "failed to send acks", e);
+      if (isAlive()) {
+        logger.log(Level.WARNING, "failed to send ack operations", e);
+      }
```
Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java]: Maybe log at a verbose level if it's not alive?

Comment (pongad) [reply google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java]: Done

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**